frontend/app/route-helpers/status-route-helpers.server.ts

```diff
@@ -0,0 +1,156 @@
+    id?: string;
+    statusId?: string | null;
+  };
+}
+
+/**
+ * Schema for validating UUID.
+ */
+const idSchema = z.string().uuid();
+
+/**
+ * Gets the session name.
+ * @param id - The ID.
+ * @returns The session name.
+ */
+function getSessionName(id: string) {
+  return `status-flow-${idSchema.parse(id)}`;
+}
+
+interface LoadStateArgs {
+  params: Params;
+  session: Session;
+}
+
+/**
+ * Loads state.
+ * @param args - The arguments.
+ * @returns The loaded state.
+ */
+export function loadStatusState({ params, session }: LoadStateArgs) {
+  const locale = getLocaleFromParams(params);
+  const cdcpWebsiteApplyUrl = getCdcpWebsiteApplyUrl(locale);
+
+  const parsedId = idSchema.safeParse(params.id);
+
+  if (!parsedId.success) {
+    log.warn('Invalid "id" param format; redirecting to [%s]; id: [%s], sessionId: [%s]', cdcpWebsiteApplyUrl, params.id, session.id);
+    throw redirectDocument(cdcpWebsiteApplyUrl);
+  }
+
+  const sessionName = getSessionName(parsedId.data);
+
+  if (!session.has(sessionName)) {
+    log.warn('Status session state has not been found; redirecting to [%s]; sessionName: [%s], sessionId: [%s]', cdcpWebsiteApplyUrl, sessionName, session.id);
+    throw redirectDocument(cdcpWebsiteApplyUrl);
+  }
+
+  const state: StatusState = session.get(sessionName);
+
+  // Checks if the elapsed time since the last update exceeds 20 minutes,
+  // and performs necessary actions if it does.
+  const lastUpdatedOn = new UTCDate(state.lastUpdatedOn);
+  const now = new UTCDate();
+
+  if (differenceInMinutes(now, lastUpdatedOn) >= 20) {
+    session.unset(sessionName);
+    log.warn('Status session state has expired; redirecting to [%s]; sessionName: [%s], sessionId: [%s]', cdcpWebsiteApplyUrl, sessionName, session.id);
+    throw redirectDocument(cdcpWebsiteApplyUrl);
+  }
+
+  return state;
+}
+
+interface SaveStateArgs {
+  params: Params;
+  session: Session;
+  state: Partial<OmitStrict<StatusState, 'id' | 'lastUpdatedOn'>>;
+  remove?: keyof OmitStrict<StatusState, 'id' | 'lastUpdatedOn'>;
+}
+
+/**
+ * Saves status state.
+ * @param args - The arguments.
+ * @returns The new status state.
+ */
+export function saveStatusState({ params, session, state, remove = undefined }: SaveStateArgs) {
+  const currentState = loadStatusState({ params, session });
+
+  let newState = {
+    ...currentState,
+    ...state,
+    lastUpdatedOn: new UTCDate().toISOString(),
+  } satisfies StatusState;
+
+  if (remove && remove in newState) {
+    newState = omit(newState, [remove]);
+  }
+
+  const sessionName = getSessionName(currentState.id);
+  session.set(sessionName, newState);
+  log.info('Status session state saved; sessionName: [%s], sessionId: [%s]', sessionName, session.id);
+  return newState;
+}
+
+interface ClearStateArgs {
+  params: Params;
+  session: Session;
+}
+
+/**
+ * Clears status state.
+ * @param args - The arguments.
+ */
+export function clearStatusState({ params, session }: ClearStateArgs) {
```

Comment:
I noticed `clearStatusState(..)` isn't currently being called anywhere. Should it be used somewhere or can it be removed :man_shrugging: ?

Reply:
Added clear states to cancel/exit buttons and error redirects